framework/src/constraints/AutomaticMortarGeneration.C

```diff
@@ -1858,6 +1861,122 @@ AutomaticMortarGeneration::projectSecondaryNodes()
     projectSecondaryNodesSinglePair(pr.first, pr.second);
 }
 
+bool
+AutomaticMortarGeneration::processAlignedNodes(
+    const Node & secondary_node,
+    const Node & primary_node,
+    const std::vector<const Elem *> * secondary_node_neighbors,
+    const std::vector<const Elem *> * primary_node_neighbors,
+    const VectorValue<Real> & nodal_normal,
+    const Elem & candidate_element,
+    std::set<const Elem *> & rejected_elem_candidates)
+{
+  if (!secondary_node_neighbors)
+    secondary_node_neighbors = &libmesh_map_find(_nodes_to_secondary_elem_map, secondary_node.id());
+  if (!primary_node_neighbors)
+    primary_node_neighbors = &libmesh_map_find(_nodes_to_primary_elem_map, primary_node.id());
+
+  std::vector<bool> primary_elems_mapped(primary_node_neighbors->size(), false);
+
+  // Add entries to secondary_node_and_elem_to_xi2_primary_elem container.
+  //
+  // First, determine "on left" vs. "on right" orientation of the nodal neighbors.
+  // There can be a max of 2 nodal neighbors, and we want to make sure that the
+  // secondary nodal neighbor on the "left" is associated with the primary nodal
+  // neighbor on the "left" and similarly for the "right". We use cross products to determine
+  // alignment. In the below diagram, 'x' denotes a node, and connected '|' are lower dimensional
+  // elements.
+  //                   x
+  //           x       |
+  //           |       |
+  // secondary x ----> x primary
+  //           |       |
+  //           |       x
+  //           x
+  //
+  //  Looking at the aligned nodes, the secondary node first, if we pick the top secondary lower
+  //  dimensional element, then the cross product as written a few lines below points out of the
+  //  screen towards you. (Point in the direction of the secondary nodal normal, and then curl your
+  //  hand towards the secondary element's opposite node, then the thumb points in the direction of
+  //  the cross product). Doing the same with the aligned primary node, if we pick the top primary
+  //  element, then the cross product also points out of the screen. Because the cross products
+  //  point in the same direction (positive dot product), then we know to associate the
+  //  secondary-primary pair. If we had picked the bottom primary element whose cross product points
+  //  into the screen, then clearly the cross products point in the opposite direction and we don't
+  //  have a match
+  std::array<Real, 2> secondary_node_neighbor_cps, primary_node_neighbor_cps;
+
+  for (const auto nn : index_range(*secondary_node_neighbors))
+  {
+    const Elem * const secondary_neigh = (*secondary_node_neighbors)[nn];
```

Comment:
```suggestion
    const Elem * const secondary_neigh = (*secondary_node_neighbors)[nn];
    mooseAssert(const secondary_neigh->first_order_equivalent_type() == EDGE2, "Should be an edge");
```
This routine is only in the context of 2D mortar right?